nemo/collections/asr/parts/submodules/subsampling.py

```diff
@@ -91,48 +93,28 @@ def __init__(self, subsampling, subsampling_factor, feat_in, feat_out, conv_chan
         else:
             raise ValueError(f"Not valid sub-sampling: {subsampling}!")
 
-        in_length = torch.tensor(feat_in, dtype=torch.int)
-        for i in range(self._sampling_num):
-            # length=int(in_length),
-            out_length = calc_length(
-                lengths=in_length,
-                padding=self._padding,
-                kernel_size=self._kernel_size,
-                stride=self._stride,
-                ceil_mode=self._ceil_mode,
-            )
-            in_length = out_length
-
-        out_length = int(out_length)
-        self.out = torch.nn.Linear(conv_channels * out_length, feat_out)
+        self.add_pad = torch.tensor((2 * self._padding) - self._kernel_size, dtype=torch.float)
```

Comment:
These parameters are the same in the normal example but may be different in the same model for different modules. Can we keep the method of calc_length with those inputs?

Reply:
Done.